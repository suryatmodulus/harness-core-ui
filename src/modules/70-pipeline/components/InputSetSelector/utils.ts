import type { IconName, SelectOption } from '@harness/uicore'
import type { EntityGitDetails, InputSetSummaryResponse } from 'services/pipeline-ng'
import css from './InputSetSelector.module.scss'

type InputSetLocal = InputSetSummaryResponse & SelectOption
export interface InputSetValue extends InputSetLocal {
  type: InputSetSummaryResponse['inputSetType']
  gitDetails?: EntityGitDetails
}

export const getIconByType = (type: InputSetSummaryResponse['inputSetType']): IconName => {
  return type === 'OVERLAY_INPUT_SET' ? 'step-group' : 'yaml-builder-input-sets'
}

export const onDragStart = (event: React.DragEvent<HTMLLIElement>, row: InputSetValue): void => {
  event.dataTransfer.setData('data', JSON.stringify(row))
  event.currentTarget.classList.add(css.dragging)
}

export const onDragEnd = (event: React.DragEvent<HTMLLIElement>): void => {
  event.currentTarget.classList.remove(css.dragging)
}

export const onDragLeave = (event: React.DragEvent<HTMLLIElement>): void => {
  event.currentTarget.classList.remove(css.dragOver)
}

export const onDragOver = (event: React.DragEvent<HTMLLIElement>): void => {
  if (event.preventDefault) {
    event.preventDefault()
  }
  event.currentTarget.classList.add(css.dragOver)
  event.dataTransfer.dropEffect = 'move'
}
