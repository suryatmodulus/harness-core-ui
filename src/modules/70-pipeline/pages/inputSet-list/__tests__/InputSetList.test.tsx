/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, RenderResult, waitFor, fireEvent, getByText, queryByText, getByTestId } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { noop } from 'lodash-es'
import { TestWrapper, findDialogContainer, findPopoverContainer } from '@common/utils/testUtils'
import routes from '@common/RouteDefinitions'
import { defaultAppStoreValues } from '@common/utils/DefaultAppStoreData'
import { accountPathProps, pipelineModuleParams, pipelinePathProps } from '@common/utils/routeUtils'
import { branchStatusMock, gitConfigs, sourceCodeManagers } from '@connectors/mocks/mock'
import { PipelineResponse as PipelineDetailsMockResponse } from '../../pipeline-details/__tests__/PipelineDetailsMocks'
import InputSetList from '../InputSetList'
import {
  TemplateResponse,
  PipelineResponse,
  ConnectorResponse,
  GetInputSetsResponse,
  GetInputSetEdit,
  MergeInputSetResponse,
  GetOverlayInputSetEdit
} from './InputSetListMocks'

jest.mock('@common/components/YAMLBuilder/YamlBuilder')

jest.mock('@common/hooks', () => ({
  ...(jest.requireActual('@common/hooks') as any),

  useMutateAsGet: jest.fn().mockImplementation(() => {
    return TemplateResponse
  })
}))
const deleteInputSetMock = jest.fn()
const deleteInputSet = (): Promise<{ status: string }> => {
  deleteInputSetMock()
  return Promise.resolve({ status: 'SUCCESS' })
}
const getInputSetList = jest.fn()

jest.mock('@common/components/YAMLBuilder/YamlBuilder')
jest.mock('@common/utils/YamlUtils', () => ({}))
jest.mock('services/cd-ng', () => ({
  useGetConnector: jest.fn(() => ConnectorResponse),
  useCreatePR: jest.fn(() => noop),
  useGetFileContent: jest.fn(() => noop),
  useGetListOfBranchesWithStatus: jest.fn().mockImplementation(() => {
    return { data: branchStatusMock, refetch: jest.fn(), loading: false }
  }),
  useListGitSync: jest.fn().mockImplementation(() => {
    return { data: gitConfigs, refetch: jest.fn() }
  }),
  useGetSourceCodeManagers: jest.fn().mockImplementation(() => {
    return { data: sourceCodeManagers, refetch: jest.fn() }
  })
}))

jest.mock('services/pipeline-ng', () => ({
  useGetPipeline: jest.fn(() => PipelineResponse),
  useGetPipelineSummary: jest.fn(() => PipelineDetailsMockResponse),
  useGetTemplateFromPipeline: jest.fn(() => TemplateResponse),
  useGetStagesExecutionList: jest.fn(() => ({})),
  useGetMergeInputSetFromPipelineTemplateWithListInput: jest.fn(() => MergeInputSetResponse),
  useGetOverlayInputSetForPipeline: jest.fn(() => GetOverlayInputSetEdit),
  useCreateInputSetForPipeline: jest.fn(() => ({})),
  useUpdateInputSetForPipeline: jest.fn(() => ({})),
  useUpdateOverlayInputSetForPipeline: jest.fn(() => ({})),
  useCreateOverlayInputSetForPipeline: jest.fn(() => ({})),
  useGetInputSetsListForPipeline: jest.fn().mockImplementation(args => {
    getInputSetList(args)
    return GetInputSetsResponse
  }),
  getInputSetForPipelinePromise: jest.fn().mockImplementation(() => Promise.resolve(GetInputSetsResponse.data)),
  useGetInputSetForPipeline: jest.fn(() => GetInputSetEdit),
  useDeleteInputSetForPipeline: jest.fn().mockImplementation(() => ({ mutate: deleteInputSet })),
  usePostPipelineExecuteWithInputSetYaml: jest.fn(() => ({})),
  useGetSchemaYaml: jest.fn(() => ({})),
  useSoftDeletePipeline: jest.fn().mockImplementation(() => ({ mutate: jest.fn() })),
  useGetInputsetYaml: jest.fn(() => ({ data: null }))
}))

const TEST_PATH = routes.toInputSetList({ ...accountPathProps, ...pipelinePathProps, ...pipelineModuleParams })

describe('Input Set List tests', () => {
  test('render Input Set List view', async () => {
    const { getAllByText, container } = render(
      <TestWrapper
        path={TEST_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={defaultAppStoreValues}
      >
        <InputSetList />
      </TestWrapper>
    )
    await waitFor(() => getAllByText('OverLayInput'))
    expect(container).toMatchSnapshot()
  })
})

describe('Input Set List - Actions tests', () => {
  let container: HTMLElement | undefined
  let getAllByText: RenderResult['getAllByText'] | undefined

  beforeEach(async () => {
    const renderObj = render(
      <TestWrapper
        path={TEST_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        // queryParams={{ repoIdentifier: 'firstRepo', branch: 'master' }}
        defaultAppStoreValues={defaultAppStoreValues}
      >
        <InputSetList />
      </TestWrapper>
    )
    container = renderObj.container
    getAllByText = renderObj.getAllByText
    await waitFor(() => getAllByText?.('OverLayInput'))
  })

  test('click handler for Input Set', async () => {
    const inputSetRow = getAllByText?.('asd')[0]
    await act(async () => {
      fireEvent.click(inputSetRow!)
      await waitFor(() => getByTestId(document.body, 'location'))
      expect(
        getByTestId(document.body, 'location').innerHTML.endsWith(
          routes.toInputSetForm({
            accountId: 'testAcc',
            orgIdentifier: 'testOrg',
            projectIdentifier: 'test',
            pipelineIdentifier: 'pipeline',
            inputSetIdentifier: 'asd',
            module: 'cd'
          } as any)
        )
      ).toBeTruthy()
    })
  })

  test('click handler for Overlay Input Set', async () => {
    const inputSetRow = getAllByText?.('OverLayInput')[0]
    await act(async () => {
      fireEvent.click(inputSetRow!)
      await waitFor(() => getByText(document.body, 'inputSets.editOverlayTitle'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
    })
  })

  test('click handler for edit Overlay Input Set from menu', async () => {
    const menu = container?.querySelectorAll("[icon='more']")[0]
    fireEvent.click(menu!)
    const editMenu = getAllByText?.('edit')[0]
    await act(async () => {
      fireEvent.click(editMenu!)
      await waitFor(() => getByText(document.body, 'inputSets.editOverlayTitle'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
    })
  })

  test('click handler for edit Input Set from menu', async () => {
    const menu = container?.querySelectorAll("[icon='more']")[1]
    fireEvent.click(menu!)
    const editMenu = getAllByText?.('edit')[0]
    await act(async () => {
      fireEvent.click(editMenu!)
      await waitFor(() => getByTestId(document.body, 'location'))
      expect(
        getByTestId(document.body, 'location').innerHTML.endsWith(
          routes.toInputSetForm({
            accountId: 'testAcc',
            orgIdentifier: 'testOrg',
            projectIdentifier: 'test',
            pipelineIdentifier: 'pipeline',
            inputSetIdentifier: 'asd',
            module: 'cd'
          } as any)
        )
      ).toBeTruthy()
    })
  })

  test('click handler for delete Overlay Input Set from menu', async () => {
    deleteInputSetMock.mockReset()
    const menu = container?.querySelectorAll("[icon='more']")[0]
    fireEvent.click(menu!)
    const deleteMenu = getAllByText?.('delete')[0]
    await act(async () => {
      fireEvent.click(deleteMenu!)
      await waitFor(() => getByText(document.body, 'delete inputSets.overlayInputSet'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
      const deleteBtn = queryByText(form as HTMLElement, 'delete')
      fireEvent.click(deleteBtn!)
      await waitFor(() => getByText(document.body, 'inputSets.inputSetDeleted'))
      expect(deleteInputSetMock).toBeCalled()
    })
  })

  test('click handler for delete Input Set from menu', async () => {
    deleteInputSetMock.mockReset()
    const menu = container?.querySelectorAll("[icon='more']")[1]
    fireEvent.click(menu!)
    const deleteMenu = getAllByText?.('delete')[0]
    await act(async () => {
      fireEvent.click(deleteMenu!)
      await waitFor(() => getByText(document.body, 'delete inputSets.inputSetLabel'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
      const deleteBtn = queryByText(form as HTMLElement, 'delete')
      fireEvent.click(deleteBtn!)
      await waitFor(() => getByText(document.body, 'inputSets.inputSetDeleted'))
      expect(deleteInputSetMock).toBeCalled()
    })
  })

  test('click handler for new Overlay Input Set', async () => {
    const menu = getAllByText?.('inputSets.newInputSet')[0]
    fireEvent.click(menu!)
    const popover = findPopoverContainer()
    const newInputSet = getByText(popover as HTMLElement, 'inputSets.overlayInputSet')
    await act(async () => {
      fireEvent.click(newInputSet)
      await waitFor(() => getByText(document.body, 'inputSets.newOverlayInputSet'))
      let form = findDialogContainer()
      expect(form).toBeTruthy()
      // Close
      fireEvent.click(form?.querySelector('[icon="small-cross"]')!)
      form = findDialogContainer()
      expect(form).not.toBeTruthy()
    })
  })

  test('click handler for new Input Set', async () => {
    const menu = getAllByText?.('inputSets.newInputSet')[0]
    fireEvent.click(menu!)
    const popover = findPopoverContainer()
    const newInputSet = getByText(popover as HTMLElement, 'inputSets.inputSetLabel')
    await act(async () => {
      fireEvent.click(newInputSet)
      await waitFor(() => getByTestId(document.body, 'location'))
      expect(
        getByTestId(document.body, 'location').innerHTML.endsWith(
          routes.toInputSetForm({
            accountId: 'testAcc',
            orgIdentifier: 'testOrg',
            projectIdentifier: 'test',
            pipelineIdentifier: 'pipeline',
            inputSetIdentifier: '-1',
            module: 'cd'
          } as any)
        )
      ).toBeTruthy()
    })
  })

  test('search Input Set list', async () => {
    getInputSetList.mockReset()
    const searchInput = container?.querySelector('[placeholder="inputSets.searchInputSet"]') as HTMLInputElement
    fireEvent.change(searchInput, { target: { value: 'asd' } })
    expect(getInputSetList).toBeCalledWith({
      debounce: 300,
      queryParams: {
        accountIdentifier: 'testAcc',
        orgIdentifier: 'testOrg',
        pageIndex: 0,
        pageSize: 20,
        pipelineIdentifier: 'pipeline',
        projectIdentifier: 'test',
        searchTerm: 'asd'
      }
    })
  })
})
