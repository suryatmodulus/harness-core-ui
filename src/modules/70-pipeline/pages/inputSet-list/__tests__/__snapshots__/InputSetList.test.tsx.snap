// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Set List tests render Input Set List view 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main PageSubHeader--container StyledProps--flex"
  >
    <div
      class="Layout--horizontal StyledProps--main"
    >
      <span
        class="bp3-popover-wrapper"
      >
        <span
          class="bp3-popover-target"
        >
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withRightIcon Button--variation Button--variation-primary"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              inputSets.newInputSet
            </span>
            <span
              class="bp3-icon bp3-icon-caret-down StyledProps--main"
              icon="caret-down"
            >
              <svg
                data-icon="caret-down"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  caret-down
                </desc>
                <path
                  d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </span>
      </span>
    </div>
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
    >
      <div
        class="TextInput--main searchWrapper"
      >
        <div
          class="bp3-input-group"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input"
            placeholder="inputSets.searchInputSet"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main PageBody--pageBody"
    data-name="page-body"
  >
    <div
      class="TableV2--table table"
    >
      <div
        class="TableV2--header"
        role="row"
      >
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 30%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            PIPELINE.INPUTSETS.INPUTSETNAMELABEL
          </p>
        </div>
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 35%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            DESCRIPTION
          </p>
        </div>
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 30%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            ACTION
          </p>
        </div>
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 5%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            
          </p>
        </div>
      </div>
      <div
        class="TableV2--body"
      >
        <div
          class="TableV2--row TableV2--card TableV2--clickable"
          role="row"
        >
          <div
            class="TableV2--cell"
            role="cell"
            style="width: 30%;"
          >
            <div
              class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
            >
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-blue500"
                data-icon="step-group"
              >
                <svg
                  height="30"
                  id="step-group_svg__Layer_1"
                  viewBox="0 0 21 20"
                  width="30"
                  x="0"
                  xml:space="preserve"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0"
                >
                  <style>
                    .step-group_svg__st0{fill:#fff!important}
                  </style>
                  <path
                    class="step-group_svg__st0"
                    d="M1 .8h13v13.3H1z"
                  />
                  <path
                    d="M14 1H1v13h13V1zm1-1v15H0V0h15z"
                  />
                  <path
                    class="step-group_svg__st0"
                    d="M7 6h13v13H7z"
                  />
                  <path
                    class="step-group_svg__st1"
                    d="M20 6H7v13h13V6zm1-1v15H6V5h15z"
                  />
                </svg>
              </span>
              <div
                class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--padding-right-medium"
                style="flex-shrink: 1;"
              >
                <div>
                  <div
                    class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                    data-testid="OverLayInput"
                  >
                    <p
                      class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black"
                      style="--text-line-clamp: 1;"
                    >
                      OverLayInput
                    </p>
                  </div>
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-grey400"
                    style="--text-line-clamp: 1;"
                  >
                    idLabel
                  </p>
                </div>
                <div
                  class="StyledProps--font StyledProps--main StyledProps--padding-left-large"
                >
                  <span
                    class="bp3-popover-wrapper bp3-dark"
                  >
                    <span
                      class="bp3-popover-target"
                    >
                      <span
                        class="bp3-icon StyledProps--main badgeIcon StyledProps--intent-danger"
                        data-icon="error-outline"
                        data-testid="invalid-icon"
                        tabindex="0"
                      >
                        <svg
                          fill="none"
                          height="20"
                          viewBox="0 0 16 16"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M7.423 1.631a.667.667 0 011.154 0l6.543 11.333a.667.667 0 01-.577 1H1.457a.667.667 0 01-.577-1L7.423 1.631z"
                            fill="#fff"
                          />
                          <path
                            clip-rule="evenodd"
                            d="M7.57 2.253a.5.5 0 01.869 0l5.993 10.5a.5.5 0 01-.434.747H2.011a.5.5 0 01-.434-.748L7.57 2.253zm.407 8.242c.422 0 .763.333.763.744a.754.754 0 01-.763.745.754.754 0 01-.763-.745c0-.411.341-.745.763-.745zm.5-1.498a.5.5 0 11-1 0V5.498a.5.5 0 111 0v3.499z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="TableV2--cell"
            role="cell"
            style="width: 35%;"
          >
            <p
              class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--padding-right-medium StyledProps--color StyledProps--color-grey400"
              style="--text-line-clamp: 2;"
            >
              OverLayInput
            </p>
          </div>
          <div
            class="TableV2--cell"
            role="cell"
            style="width: 30%;"
          >
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-success Button--with-current-color Button--withLeftIcon Button--variation Button--variation-primary"
              type="button"
            >
              <span
                class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                data-icon="run-pipeline"
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                >
                  <defs>
                    <path
                      d="M7.2 10.6V5.4L10.6 8l-3.4 2.6zm4.4-3.3L7.4 4.1c-.3-.2-.6-.2-.9-.1-.3.2-.5.5-.5.8v6.4c0 .3.2.6.5.8.3.1.6.1.9-.1l4.2-3.2c.2-.2.3-.4.3-.7s-.1-.5-.3-.7zm1.1 5.4c-2.2 2.2-5.7 2.6-8.4.8s-3.6-5.1-2.4-8c1.2-2.9 4.3-4.6 7.4-4 3.1.6 5.3 3.3 5.3 6.5.1 1.8-.6 3.4-1.9 4.7zm2.7-7.8C13.8 1.1 9.6-.9 5.7.4 1.7 1.6-.7 5.6.2 9.6s4.5 6.8 8.6 6.4c4.1-.4 7.2-3.8 7.2-8 0-1.1-.2-2.1-.6-3.1z"
                      id="run-pipeline_svg__a"
                    />
                  </defs>
                  <clippath
                    id="run-pipeline_svg__b"
                  >
                    <use
                      overflow="visible"
                      xlink:href="#run-pipeline_svg__a"
                    />
                  </clippath>
                  <path
                    clip-path="url(#run-pipeline_svg__b)"
                    d="M-5-5h26v26H-5z"
                    fill="#54b610"
                  />
                </svg>
              </span>
              <span
                class="bp3-button-text"
              >
                runPipeline
              </span>
            </button>
          </div>
          <div
            class="TableV2--cell"
            role="cell"
            style="width: 5%;"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="justify-content: flex-end;"
            >
              <span
                class="bp3-popover-wrapper bp3-dark"
              >
                <span
                  class="bp3-popover-target"
                >
                  <button
                    class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main actionButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                    type="button"
                  >
                    <span
                      class="bp3-icon bp3-icon-more StyledProps--main"
                      icon="more"
                    >
                      <svg
                        data-icon="more"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                      >
                        <desc>
                          more
                        </desc>
                        <path
                          d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="TableV2--row TableV2--card TableV2--clickable"
          role="row"
        >
          <div
            class="TableV2--cell"
            role="cell"
            style="width: 30%;"
          >
            <div
              class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
            >
              <span
                class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-black"
                data-icon="yaml-builder-input-sets"
              >
                <svg
                  height="30"
                  viewBox="0 0 18 16"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="#22272D"
                  >
                    <path
                      d="M12.946 5.347a.49.49 0 00-.492-.486H4.581a.49.49 0 00-.492.486c0 .269.22.486.492.486h7.873a.49.49 0 00.492-.486zM4.581 8.264c-.272 0-.492.218-.492.486s.22.486.492.486H9.01c.271 0 .492-.218.492-.486s-.22-.486-.492-.486H4.58zm7.92 5.833H3.644a.55.55 0 01-.538-.56V2.018a.55.55 0 01.538-.56h9.748a.55.55 0 01.54.56v5.325c0 .269.22.486.491.486a.49.49 0 00.492-.486V2.02c0-.844-.679-1.533-1.523-1.533H3.643c-.844 0-1.523.69-1.523 1.533v11.517c0 .844.68 1.533 1.523 1.533h8.859a.49.49 0 00.492-.486.49.49 0 00-.492-.486z"
                    />
                    <path
                      d="M15.676 10.418h-1.005V9.413a.524.524 0 10-1.048 0v1.004H12.28a.524.524 0 000 1.049h1.343v1.343a.524.524 0 101.048 0v-1.343h1.005a.524.524 0 000-1.048z"
                    />
                  </g>
                </svg>
              </span>
              <div
                class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--padding-right-medium"
                style="flex-shrink: 1;"
              >
                <div>
                  <div
                    class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                    data-testid="asd"
                  >
                    <p
                      class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black"
                      style="--text-line-clamp: 1;"
                    >
                      asd
                    </p>
                  </div>
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-grey400"
                    style="--text-line-clamp: 1;"
                  >
                    idLabel
                  </p>
                </div>
                <div
                  class="StyledProps--font StyledProps--main StyledProps--padding-left-large"
                >
                  <span
                    class="bp3-popover-wrapper bp3-dark"
                  >
                    <span
                      class="bp3-popover-target"
                    >
                      <span
                        class="bp3-icon StyledProps--main badgeIcon StyledProps--intent-danger"
                        data-icon="error-outline"
                        data-testid="invalid-icon"
                        tabindex="0"
                      >
                        <svg
                          fill="none"
                          height="20"
                          viewBox="0 0 16 16"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M7.423 1.631a.667.667 0 011.154 0l6.543 11.333a.667.667 0 01-.577 1H1.457a.667.667 0 01-.577-1L7.423 1.631z"
                            fill="#fff"
                          />
                          <path
                            clip-rule="evenodd"
                            d="M7.57 2.253a.5.5 0 01.869 0l5.993 10.5a.5.5 0 01-.434.747H2.011a.5.5 0 01-.434-.748L7.57 2.253zm.407 8.242c.422 0 .763.333.763.744a.754.754 0 01-.763.745.754.754 0 01-.763-.745c0-.411.341-.745.763-.745zm.5-1.498a.5.5 0 11-1 0V5.498a.5.5 0 111 0v3.499z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="TableV2--cell"
            role="cell"
            style="width: 35%;"
          >
            <p
              class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--padding-right-medium StyledProps--color StyledProps--color-grey400"
              style="--text-line-clamp: 2;"
            >
              asd
            </p>
          </div>
          <div
            class="TableV2--cell"
            role="cell"
            style="width: 30%;"
          >
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-success Button--with-current-color Button--withLeftIcon Button--variation Button--variation-primary"
              type="button"
            >
              <span
                class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                data-icon="run-pipeline"
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                >
                  <defs>
                    <path
                      d="M7.2 10.6V5.4L10.6 8l-3.4 2.6zm4.4-3.3L7.4 4.1c-.3-.2-.6-.2-.9-.1-.3.2-.5.5-.5.8v6.4c0 .3.2.6.5.8.3.1.6.1.9-.1l4.2-3.2c.2-.2.3-.4.3-.7s-.1-.5-.3-.7zm1.1 5.4c-2.2 2.2-5.7 2.6-8.4.8s-3.6-5.1-2.4-8c1.2-2.9 4.3-4.6 7.4-4 3.1.6 5.3 3.3 5.3 6.5.1 1.8-.6 3.4-1.9 4.7zm2.7-7.8C13.8 1.1 9.6-.9 5.7.4 1.7 1.6-.7 5.6.2 9.6s4.5 6.8 8.6 6.4c4.1-.4 7.2-3.8 7.2-8 0-1.1-.2-2.1-.6-3.1z"
                      id="run-pipeline_svg__a"
                    />
                  </defs>
                  <clippath
                    id="run-pipeline_svg__b"
                  >
                    <use
                      overflow="visible"
                      xlink:href="#run-pipeline_svg__a"
                    />
                  </clippath>
                  <path
                    clip-path="url(#run-pipeline_svg__b)"
                    d="M-5-5h26v26H-5z"
                    fill="#54b610"
                  />
                </svg>
              </span>
              <span
                class="bp3-button-text"
              >
                runPipeline
              </span>
            </button>
          </div>
          <div
            class="TableV2--cell"
            role="cell"
            style="width: 5%;"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="justify-content: flex-end;"
            >
              <span
                class="bp3-popover-wrapper bp3-dark"
              >
                <span
                  class="bp3-popover-target"
                >
                  <button
                    class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main actionButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                    type="button"
                  >
                    <span
                      class="bp3-icon bp3-icon-more StyledProps--main"
                      icon="more"
                    >
                      <svg
                        data-icon="more"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                      >
                        <desc>
                          more
                        </desc>
                        <path
                          d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Layout--horizontal StyledProps--main TableV2--pagination Pagination--container StyledProps--padding-top-medium StyledProps--padding-bottom-medium StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-distribution-space-between"
      >
        <span
          class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-small-semi"
        >
          3
           of 
          3
        </span>
        <div
          class="Layout--horizontal StyledProps--main"
        >
          <button
            class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--buttonLeft Button--with-current-color Button--withLeftIcon Button--size Button--small"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-arrow-left StyledProps--main StyledProps--padding-right-xsmall"
              icon="arrow-left"
            >
              <svg
                data-icon="arrow-left"
                height="12"
                viewBox="0 0 16 16"
                width="12"
              >
                <desc>
                  arrow-left
                </desc>
                <path
                  d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="bp3-button-text"
            >
              Prev
            </span>
          </button>
          <button
            class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--selected Button--with-current-color"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              1
            </span>
          </button>
          <button
            class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--buttonRight Button--with-current-color Button--withRightIcon Button--size Button--small"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Next
            </span>
            <span
              class="bp3-icon bp3-icon-arrow-right StyledProps--main"
              icon="arrow-right"
            >
              <svg
                data-icon="arrow-right"
                height="12"
                viewBox="0 0 16 16"
                width="12"
              >
                <desc>
                  arrow-right
                </desc>
                <path
                  d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
        <span>
          Showing 10 per page
        </span>
      </div>
    </div>
  </div>
</div>
`;
